src/app/clusters/scenes/ExtensionFieldsSetsImpl.cpp

```diff
@@ -0,0 +1,151 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/clusters/scenes/ExtensionFieldsSetsImpl.h>
+
+namespace chip {
+namespace scenes {
+
+ExtensionFieldsSetsImpl::ExtensionFieldsSetsImpl() : ExtensionFieldsSets() {}
+
+CHIP_ERROR ExtensionFieldsSetsImpl::Serialize(TLV::TLVWriter & writer) const
+{
+    TLV::TLVType container;
+    ReturnErrorOnFailure(writer.StartContainer(TLV::ContextTag(1), TLV::kTLVType_Structure, container));
+    ReturnErrorOnFailure(writer.Put(TagFieldNum(), static_cast<uint8_t>(this->mFieldNum)));
+    if (!this->is_empty())
+    {
+        for (uint8_t i = 0; i < kMaxClusterPerScenes; i++)
+        {
+            if (!this->mEFS[i].is_empty())
+            {
+                LogErrorOnFailure(this->mEFS[i].Serialize(writer));
```

Comment:
Logging only on failure may cause the scenes to be improperly recorded. We should return error on serialize, so the cluster returns an error to the client. Otherwise we could have bad partial scenes storage AND the client/application will not be able to react. On production software, it has to be assumed that logs go basically nowhere.

Reply:
Fixed in latest commit